Lib/test/test_compileall.py

```diff
@@ -55,9 +55,21 @@ def timestamp_metadata(self):
         with open(self.bc_path, 'rb') as file:
             data = file.read(12)
         mtime = int(os.stat(self.source_path).st_mtime)
-        compare = struct.pack('<4sll', importlib.util.MAGIC_NUMBER, 0, mtime)
+        compare = struct.pack('<4sLL', importlib.util.MAGIC_NUMBER, 0,
+                              mtime & 0xFFFF_FFFF)
         return data, compare
 
+    def test_year_2038_mtime_compilation(self):
+        # Test to make sure we can handle mtimes larger than what a 32-bit
+        # signed number can hold as part of bpo-34990
+        with open(self.source_path, 'r') as f:
+            os.utime(f.name, (2**32 - 1, 2**32 - 1))
+        self.assertTrue(compileall.compile_file(self.source_path))
+
+        with open(self.source_path, 'r') as f:
+            os.utime(f.name, (2**35, 2**35))
```

Comment:
I would catch OverflowError and OSError and skip the test. It is better to split this test on two parts, because only part of it can be skipped.